In [1]:
(defvar p-cancer 0.001)

P-CANCER

In [2]:
(defvar p-not-cancer .999)

P-NOT-CANCER

In [3]:
(defvar p-test-pos-cancer 0.8)

P-TEST-POS-CANCER

In [4]:
(defvar p-test-pos-not-cancer 0.1)

P-TEST-POS-NOT-CANCER

In [5]:
(defvar p-test-pos (+ (* p-test-pos-cancer p-cancer)
                      (* p-test-pos-not-cancer p-not-cancer)))

P-TEST-POS

In [6]:
p-test-pos

0.1007

In [7]:
(defun bayes (A B BA)
    (/ (* BA A) B))

BAYES

In [8]:
(bayes p-cancer 
       p-test-pos
       p-test-pos-cancer)

0.00794439

**Despite having a positive test for cancer, the probability of actually having cancer is still only 0.79%**